src/main/drivers/pwm_output.c

```diff
@@ -328,12 +315,21 @@ void motorDevInit(const motorDevConfig_t *motorConfig, uint16_t idlePulse, uint8
         IOConfigGPIO(motors[motorIndex].io, IOCFG_AF_PP);
 #endif
 
+        /* standard PWM outputs */
         if (useUnsyncedPwm) {
-            const uint32_t hz = timerMhzCounter * 1000000;
-            pwmOutConfig(&motors[motorIndex], timerHardware, timerMhzCounter, hz / motorConfig->motorPwmRate, idlePulse, motorConfig->motorPwmInversion);
-        } else {
-            pwmOutConfig(&motors[motorIndex], timerHardware, timerMhzCounter, 0xFFFF, 0, motorConfig->motorPwmInversion);
+            pwmRate = motorConfig->motorPwmRate;
         }
+        const uint32_t clock = timerClock(timerHardware->tim);
+        const uint16_t prescaler = ceilf((clock / pwmRate) / 65535.0f); /* used to find the desired timer frequency for max resolution */
+        const uint32_t hz = clock / prescaler;
+        const uint16_t period = hz / pwmRate; 
+        
+        const float ticksPerUs = (float)hz / 1000000; /* aka mhz */
+        motors[motorIndex].pulseMin = lrintf(usMin * ticksPerUs);
+        // if brushed then it is the entire length of the period.
+        motors[motorIndex].pulseLen = (motorConfig->motorPwmProtocol == PWM_TYPE_BRUSHED) ? period : lrintf(usLen * ticksPerUs);
+        
+        pwmOutConfig(&motors[motorIndex], timerHardware, hz / 1000000, period, idlePulse, motorConfig->motorPwmInversion);
```

Comment:
calculation above expect best possible `hz`, but rounds it here to MHz - this multiwii heritage is terrible and may break timing now ..